### PR TITLE
Fix regression around pasting links

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/22117

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix regression around pasting links ([\#8537](https://github.com/matrix-org/matrix-react-sdk/pull/8537)). Fixes vector-im/element-web#22117.

### CODE REVIEW

File: src/components/views/rooms/BasicMessageComposer.tsx [hunk 1]
```diff
@@ -333,28 +333,29 @@ export default class BasicMessageEditor extends React.Component<IProps, IState>
...
+            parts = parsePlainTextMessage(plainText, partCreator, { shouldEscape: false });
         }
-        const textToInsert = event.clipboardData.getData("text/plain");
+
         this.modifiedFlag = true;
         const range = getRangeForSelection(this.editorRef.current, model, document.getSelection());
-        if (textToInsert && linkify.test(textToInsert)) {
-            formatRangeAsLink(range, textToInsert);
+
+        if (plainText && range.length > 0 && linkify.test(plainText)) {
```
Comment (t3chguy) [inline src/components/views/rooms/BasicMessageComposer.tsx:357]: this `range.length > 0` is the main fix

File: test/components/views/rooms/BasicMessageComposer-test.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            clipboardData: {
+                getData: type => {
+                    if (type === "text/plain") {
+                        return "https://element.io";
+                    }
+                },
+            },
+        });
+
+        expect(model.parts[0].text).toBe("https://element.io");
```
Comment (robintown) [inline test/components/views/rooms/BasicMessageComposer-test.tsx:50]: Could we also assert that there are no other parts?

File: test/editor/operations-test.ts [hunk 3]
```diff
@@ -17,21 +17,89 @@ limitations under the License.
...
+    });
+
+    describe("formatRangeAsLink", () => {
+        it.each([
+            // Caret is denoted by | in the expectation string
+            ["testing", "[testing](|)", ""],
+            ["testing", "[testing](foobar|)", "foobar"],
+            ["[testing]()", "testing|", ""],
+            ["[testing](foobar)", "testing|", ""],
+        ])("%s -> %s", (input: string, expectation: string, text: string) => {
```
Comment (robintown) [inline test/editor/operations-test.ts:86]: Could this be named a little more explicitly, like "converts ___ to ___"?

**Review by robintown — APPROVED** [review]
Looks good otherwise barring type checks

**Pull request merged**